### DISCUSSION TITLE
Keyword Boosting doesn't work on enhanced teir

### DISCUSSION THREAD

**Discussion by CharlieW12345** [deepgram/community#119] [General help] [answered]
Hi, 
I have recently been experimenting with the keyword boosting feature and have found out it doesn't work when the enhanced tier is enabled, is this meant to be the case ?

**Comment by SandraRodgers** [comment]
Hi @CharlieW12345 ,

Are you using one of our SDKs? If so, could you let me know which one? 

Keywords boosting does work with our enhanced tier. If you could provide me with a little more information about what you are doing, and whether you are using an SDK or not, I can try and reproduce the issue and find out from one of our engineers why it's not working.

Sandra

**Reply by CharlieW12345** [reply]
Hi Sandra,

I am not using an SDK, I am just making curl requests with the parameters, model=general, teir=enhanced, keywrods= etc...
So for example the URL would be https://api.deepgram.com/v1/listen?model=general&tier=enhanced&keywords=keyword:6.
When I make this request with the enhanced tier the keyword boosting does not work but when this parameter is taken away the keyword boosting works.

Charlie.

**Reply by SandraRodgers** [reply]
Hi Charlie.... I apologize. I believe you are right. After some testing and talking to others in the company, I found out that Keywords is not working for enhanced. It's currently only working with base. 

Can I ask what you are trying to do with keywords? I might be able to suggest a different feature that could work to help improve your transcription.

Sandra

**Reply by CharlieW12345** [reply]
Hi Sandra, 

I am currently trying to improve lecture recordings WER, and was essentially feeding keywords selected in different ways into the keyword boosting feature in order to do this. 

Charlie.

**Reply by SandraRodgers** [reply]
Hi Charlie,

I'm curious about this. Are you selecting the keywords manually based on the topic of the lecture? What do you mean by "selected in different ways"?

Sandra

**Reply by CharlieW12345** [reply]
Hi Sandra,

My first method is to just do a unigram count vectorization of a transcript produced by the enhanced model. Then my second method would be to build a knowledge graph from a collection of lectures that are about the same topic and preferably from the same lecturer. From this knowledge graph constructed a sub-graph can then be made to extract keywords, that hopefully contain more context than the count vectorization method. I've thought of another method that could also be developed which entails, using the search function and searching for a keyword (selected using one of the above methods), then depending on the confidence of the returned searches I would then replace them with the keyword using the replace function, if the confidence is above a certain threshold. 

Charlie.

**Reply by SandraRodgers** [reply, marked as answer]
Hi Charlie,

It sounds like you came up with a really good alternative solution. Using the search feature to search for a word and using the confidence to decide if you will replace the word is probably a better way to do this. Our keywords feature is less reliable than our search feature (we are working to improve it but it will be a while before we push out any big improvements). 

For anyone who comes across this thread and would like more info on best use cases for each feature (search vs keywords), we have this guide in our documentation https://developers.deepgram.com/documentation/guides/keywords-vs-search/

I hope you're finding this method works well for you.

Sandra
